pkg/audit/manager.go

```diff
@@ -761,11 +792,27 @@ func (am *Manager) addAuditResponsesToUpdateLists(
 		}
 
 		totalViolationsPerEnforcementAction[ea]++
+		if enableBatch {
+			batch = append(batch, violationMsg(r.Constraint, ea, gvk, namespace, name, r.Msg, details, r.obj.GetLabels(), timestamp))
+		} else {
+			err := am.pubsubSystem.Publish(context.Background(), *auditConnection, *auditChannel, violationMsg(r.Constraint, ea, gvk, namespace, name, r.Msg, details, r.obj.GetLabels(), timestamp))
+			if err != nil {
+				errs = append(errs, err)
+			}
+		}
 		logViolation(am.log, r.Constraint, ea, gvk, namespace, name, r.Msg, details, r.obj.GetLabels())
 		if *emitAuditEvents {
 			emitEvent(r.Constraint, timestamp, ea, gvk, namespace, name, rv, r.Msg, am.gkNamespace, uid, am.eventRecorder)
 		}
 	}
+	if enableBatch {
+		err := am.pubsubSystem.PublishBatch(context.Background(), *auditConnection, *auditChannel, batch)
+		if err != nil {
+			return err
```

Comment:
This implies that messages are lost if there is an error publishing them. How reliable does this channel need to be? Do we ever attempt to retry sending? Are retries handled by the call to Publish()/PublishBatch()?

If there are errors sending messages, how are users made aware that their pipeline has un-handled events?

Reply:
resolving this conversation in favor of raising a follow up PR for batch publish changes